libgnucash/engine/Account.cpp

```diff
@@ -6229,25 +6229,19 @@ gboolean xaccAccountRegister (void)
     return qof_object_register (&account_object_def);
 }
 
-
 using AccountSet = std::unordered_set<Account*>;
-static void maybe_add_descendants (AccountSet& accset, Account* acc)
+static void maybe_add_descendants (Account* acc, gpointer arg)
 {
-    if (accset.insert (acc).second)
-        for (auto m = GET_PRIVATE(acc)->children; m; m = g_list_next (m))
-            maybe_add_descendants (accset, static_cast<Account*>(m->data));
+    if (static_cast <AccountSet*> (arg)->insert (acc).second)
+        g_list_foreach (GET_PRIVATE(acc)->children, (GFunc) maybe_add_descendants, arg);
 };
 
 GList *
 gnc_accounts_and_all_descendants (GList *accounts)
 {
     AccountSet accset;
-    for (GList *n = accounts; n; n = g_list_next (n))
-        maybe_add_descendants (accset, static_cast<Account*>(n->data));
-    GList *ret = nullptr;
-    std::for_each (accset.begin(), accset.end(),
-                   [&ret](const auto& it) { ret = g_list_prepend (ret, it); });
-    return ret;
+    g_list_foreach (accounts, (GFunc) maybe_add_descendants, &accset);
+    return std::accumulate (accset.begin(), accset.end(), (GList*) nullptr, g_list_prepend);
```

Comment:
Include what you use: Add `#include <algorithm>` at the top for `std::accumulate`.

Reply:
Apparently you need a little background on Include What You Use. [Read this](https://www.fluentcpp.com/2021/01/01/include-what-you-use/).